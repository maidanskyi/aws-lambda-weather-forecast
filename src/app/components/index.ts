export * from './weatherForecast';
export * from './weatherMetrics';
