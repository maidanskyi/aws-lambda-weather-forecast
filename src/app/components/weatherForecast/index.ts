export * from './weatherForecast.controller';
export * from './weatherForecast.service';
