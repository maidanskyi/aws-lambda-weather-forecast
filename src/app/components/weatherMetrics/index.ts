export * from './weatherMetrics.controller';
export * from './weatherMetrics.service';
