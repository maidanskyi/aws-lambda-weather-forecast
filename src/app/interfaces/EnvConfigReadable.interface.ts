import { LogLevel } from '../types';

export interface EnvConfigReadable {
  getPort(): number
  getWeatherApiKeyAttributeName(): string
  getWeatherSecrets(): string
  getTableName(): string
  getLogLevel(): LogLevel
  getAwsRegion(): string
  getSecretId(): string
  getNodeEnv(): string
  isProdEnv(): boolean
  isDevEnv(): boolean
  isTestEnv(): boolean
}
