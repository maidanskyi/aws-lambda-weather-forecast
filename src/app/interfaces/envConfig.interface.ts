export interface EnvConfigReadable {
  getPort(): number
  getNodeEnv(): string
  isProdEnv(): boolean
  isDevEnv(): boolean
  isTestEnv(): boolean
}
