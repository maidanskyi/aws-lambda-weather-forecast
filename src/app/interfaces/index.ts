export * from './envConfig.interface';
export * from './logger.interface';
export * from './metric.interface';
export * from './secret.interface';
