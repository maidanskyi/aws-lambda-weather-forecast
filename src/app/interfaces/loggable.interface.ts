/* eslint-disable no-unused-vars */

interface LogFn {
  <T extends Record<string, unknown>>(msg: string, obj?: T): void;
}

export interface Loggable {
  error: LogFn
  warn: LogFn
  info: LogFn
  debug: LogFn
}
