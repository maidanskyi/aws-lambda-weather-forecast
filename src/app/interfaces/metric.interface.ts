export interface MetricPublishable {
  publishSuccessMetric(): Promise<void>
  publishFailureMetric(): Promise<void>
}
