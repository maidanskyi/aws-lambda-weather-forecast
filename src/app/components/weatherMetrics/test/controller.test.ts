import 'reflect-metadata';
import supertest from 'supertest';
import { InversifyExpressServer } from 'inversify-express-utils';

import '../../../components';
import { container } from '../../../ioc';
import { TYPES } from '../../../constants';
import { EnvConfigReadable, MetricGettable } from '../../../interfaces';


describe('', () => {
  let response: supertest.Response;
  let getMetricsSpy: jest.SpyInstance;

  beforeAll(async () => {
    const envConfigService = container.get<EnvConfigReadable>(TYPES.envConfig);

    jest
      .spyOn(envConfigService, 'getAwsRegion')
      .mockImplementation(() => {
        return 'us-east-1';
      });

    container
      .rebind(TYPES.envConfig)
      .toConstantValue(envConfigService);

    const weatherMetricService = container
      .get<MetricGettable>(TYPES.weatherMetricsService);

    getMetricsSpy = jest
      .spyOn(weatherMetricService, 'getMetrics')
      .mockImplementation(async () => {
        return ['X-Header-Prometheus', 'PROMETHEUS-METRICS'];
      });

    container
      .rebind(TYPES.weatherMetricsService)
      .toConstantValue(weatherMetricService);

    const server = new InversifyExpressServer(container).build();

    response = await supertest(server).get('/weather-metrics');
  });

  it('should call mocked getMetrics method', () => {
    expect(getMetricsSpy).toHaveBeenCalled();
  });

  it('Content-Type should be equal X-Header-Prometheus', () => {
    expect(response.headers['content-type']).toMatchSnapshot();
  });

  it('response text should be PROMETHEUS-METRICS', () => {
    expect(response.text).toMatchSnapshot();
  });
});
