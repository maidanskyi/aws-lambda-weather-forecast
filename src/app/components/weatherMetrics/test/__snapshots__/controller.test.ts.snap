// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Content-Type should be equal X-Header-Prometheus 1`] = `"X-Header-Prometheus"`;

exports[` response text should be PROMETHEUS-METRICS 1`] = `"PROMETHEUS-METRICS"`;
